library/proc_macro/src/bridge/client.rs

```diff
@@ -430,6 +411,10 @@ fn run_client<A: for<'a, 's> DecodeMut<'a, 's, ()>, R: Encode<()>>(
                 // at the moment, so this is also potentially preventing UB).
                 buf.clear();
                 Ok::<_, ()>(output).encode(&mut buf, &mut ());
+
+                // Now that the response has been serialized, invalidate all
+                // symbols registered with the interner.
+                Symbol::invalidate_all();
```

Comment:
This also needs to be done in case of panic (in the `unwrap_or_else` below) and probably even before calling `f` above, just to make sure it starts off empty.